src/distPlus/symbolic/SymbolicDist.re

```diff
@@ -296,103 +313,407 @@ module GenericSimple = {
+        let y = GenericSimple.pdf(v1, d2);
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.make({xs: [|v1|], ys: [|y|]}), y)
+    }
+    | (`Simple(d1), `Simple(`Float(v2))) => {
+        // evaluate d1 at v2
+        let y = GenericSimple.pdf(v2, d1);
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.make({xs: [|v2|], ys: [|y|]}), y)
+    }
+    | (`Simple(`Normal(n1)), `Simple(`Normal(n2))) => {
+        let mean = (n1.mean *. n2.stdev**2. +. n2.mean *. n1.stdev**2.) /. (n1.stdev**2. +. n2.stdev**2.);
+        let stdev = 1. /. ((1. /. n1.stdev**2.) +. (1. /. n2.stdev**2.));
+        let integral = 0; // TODO
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+    }
+    /* General cases */
+    | (`Simple(d1), `Simple(d2)) => {
+        // NOT IMPLEMENTED YET
+        // TODO: evaluate integral properly
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        let (sc2, sd2) = renderDistributionToXYShape(d2, n);
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+    }
+    | (`Simple(d1), `RenderedShape(sc2, sd2, i2)) => {
+        // NOT IMPLEMENTED YET
+        // TODO: evaluate integral properly
+        let (sc1, sd1) = renderDistributionToXYShape(d1, n);
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+    }
+    | (`RenderedShape(sc1, sd1, i1), `Simple(d1)) => {
+        // NOT IMPLEMENTED YET
+        // TODO: evaluate integral properly
+        let (sc2, sd2) = renderDistributionToXYShape(d1, n);
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+    }
+    | (`RenderedShape(sc1, sd1, i1), `RenderedShape(sc2, sd2, i2)) => {
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+    }
+    }
+  };
+
+
+  let evaluateNormalize = (et: nodeResult, n: int) => {
+    // just divide everything by the integral.
+    switch (et) {
+    | `RenderedShape(sc, sd, 0.) => {
+        `RenderedShape(Distributions.Continuous.empty, Distributions.Discrete.empty, 0.)
+      }
+    | `RenderedShape(sc, sd, i) => {
+        // loop through all ys and divide them by i
+        let normalize = (s: DistTypes.xyShape): DistTypes.xyShape => {xs: s.xs, ys: s.ys |> E.A.fmap(y => y /. i)};
+
+        let scn = sc |> Distributions.Continuous.shapeMap(normalize);
+        let sdn = sd |> normalize;
+
+        `RenderedShape(scn, sdn, 1.)
+    }
+    | `Simple(d) => `Simple(d) // any kind of atomic dist should already be normalized -- TODO: THIS IS ACTUALLY FALSE! E.g. pointwise product of normal * normal
+    }
+  };
+
+  let evaluateTruncate = (et: nodeResult, xc: cutoffX, compareFunc: (float, float) => bool, n: int) => {
+    let cut = (s: DistTypes.xyShape): DistTypes.xyShape => {
+      let (xs, ys) = s.ys
+        |> Belt.Array.zip(s.xs)
+        |> E.A.filter(((x, y)) => compareFunc(x, xc))
```

Comment:
This seems too simple to me. If we have xy points at [300,1] and [400,2], and want to cut this off at 350 and over, it should make a new point close to 350 using interpolation, instead of just filtering for the previous node.

Likewise, I imagine it would be messy to come up with a generic interpolation scheme, so would recommend that instead of passing in a ``compareFun`` function, we restrict this to a shorter list of operations, like,
``trunctateType = [`TrunctateGreaterThan | `TrunctateLessThan`]